gsy_framework/sim_results/electric_blue/aggregate_results.py

```diff
@@ -0,0 +1,145 @@
+# what aggregations are needed for each market type.
+MARKET_RESOLUTIONS = {
+    AvailableMarketTypes.YEAR_FORWARD: [AggregationResolution.RES_1_MONTH],
+    AvailableMarketTypes.MONTH_FORWARD: [AggregationResolution.RES_1_DAY],
+    AvailableMarketTypes.WEEK_FORWARD: [AggregationResolution.RES_1_DAY]
+}
+
+
+@dataclass
+class ForwardDeviceStats:  # pylint: disable=too-many-instance-attributes
+    """Hold forward device statistics for each timeslot. This class is used to first
+    calculate the current results of device in the market and then used to merge it
+    with the global results."""
+
+    timeslot: str
+    device_uuid: str
+    current_timeslot: DateTime
+
+    # SELLER
+    total_energy_produced: float = 0
+    total_sell_trade_count: int = 0
+    total_energy_sold: float = 0
+    total_earned_eur: float = 0
+
+    # BUYER
+    total_energy_consumed: float = 0
+    total_buy_trade_count: int = 0
+    total_energy_bought: float = 0
+    total_spent_eur: float = 0
+
+    open_offers: List[Dict] = field(default_factory=list)
+    open_bids: List[Dict] = field(default_factory=list)
+    trades: List[Dict] = field(default_factory=list)
+
+    def __add__(self, other: "ForwardDeviceStats"):
+        assert self.timeslot == other.timeslot
+        assert self.device_uuid == other.device_uuid
+
+        if self.current_timeslot >= other.current_timeslot:
+            current_timeslot = self.current_timeslot
+            open_bids = self.open_bids
+            open_offers = self.open_offers
+        else:
+            current_timeslot = other.current_timeslot
+            open_bids = other.open_bids
+            open_offers = other.open_offers
+
+        return ForwardDeviceStats(
+            timeslot=self.timeslot,
+            device_uuid=self.device_uuid,
+            current_timeslot=current_timeslot,
+            total_energy_produced=self.total_energy_produced + other.total_energy_produced,
+            total_sell_trade_count=self.total_sell_trade_count + other.total_sell_trade_count,
+            total_energy_sold=self.total_energy_sold + other.total_energy_sold,
+            total_earned_eur=self.total_earned_eur + other.total_earned_eur,
+            total_energy_consumed=self.total_energy_consumed + other.total_energy_consumed,
+            total_buy_trade_count=self.total_buy_trade_count + other.total_buy_trade_count,
+            total_energy_bought=self.total_energy_bought + other.total_energy_bought,
+            total_spent_eur=self.total_spent_eur + other.total_spent_eur,
+            open_bids=open_bids,
+            open_offers=open_offers,
+            trades=self.trades + other.trades
+        )
+
+    def to_dict(self) -> Dict:
+        """Generate a dictionary for saving the data into DB."""
+        return {f.name: getattr(self, f.name) for f in fields(self)}
+
+    def add_trade(self, trade: Dict) -> None:
+        """Add trade information to device stats."""
+        assert trade["timeslot"] == self.timeslot
```

Comment:
```suggestion
        assert trade["time_slot"] == self.timeslot
```
Here is a typo probably. There are a few below as well.

Reply:
Thanks for mentioning. I fixed them all.